### ISSUE TITLE
Mucking with sys.path breaks checking imports

### ISSUE THREAD

**Issue by dmtucker** [python/mypy#13701]
**Bug Report**

If the first 2 entries of `sys.path` are pwd, `mypy` seems to miss type checks in imported modules.

**To Reproduce**

`mypy==0.971`

``` python
# sscce.py
import sys
import mypy.api
sys.path.insert(0, sys.path[0])
print(mypy.api.run(["good.py"])[0])
```
``` python
# good.py
import bad
```
``` python
# bad.py 
foo: str = 5
```

**Expected Behavior**

```
bad.py:1: error: Incompatible types in assignment (expression has type "int", variable has type "str")
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**

```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 0.971

This issue broke `pytest-mypy` tests which use [`pytester.runpytest_subprocess`](https://docs.pytest.org/en/latest/reference/reference.html#pytest.Pytester.runpytest_subprocess) to run `pytest` in a subprocess (see https://github.com/dbader/pytest-mypy/issues/139).

**Comment by dmtucker** [comment]
I bisected `v0.961..v0.971` and found the failure first reproduces in https://github.com/python/mypy/commit/2132036a019ed26d77389c0a9af9b928911a5846, and it still reproduces on `master`.

**Comment by dmtucker** [comment]
> I'm curious if it's easy for you to work around? Note that if you explicitly ask to check bad.py you should get errors from it.

I will probably just xfail the test on any affected mypy versions. The test checks that the plugin behaves correctly when a type check fails in an imported module that wasn't collected by pytest, so checking bad.py directly wouldn't really make sense.

> Also note that there are many things you could do to sys.path that would make mypy.api.run very unhappy ;-)

That's fair. I'm not sure why cwd ends up at the front of `sys.path` twice in `runpytest_subprocess` calls. :thinking: 

>  you'd get inconsistent results between the equivalent of `python -m mypy ...` and `python -m mypy --python-executable $(which python)`

Interesting, yes indeed. pytest-mypy only relies on `mypy.api.run` FWIW.
